src/main/java/org/spongepowered/common/mixin/core/world/MixinExplosion.java

```diff
@@ -383,7 +383,7 @@ public void doExplosionB(boolean spawnParticles) {
                         if (peek.state.requiresBlockPosTracking()) {
                             peek.context.getCaptureBlockPos().setPos(blockpos);
                         }
-                        block.dropBlockAsItemWithChance(this.world, blockpos, this.world.getBlockState(blockpos), 1.0F / this.size, 0);
+                        block.dropBlockAsItemWithChance(this.world, blockpos, iblockstate, 1.0F / this.size, 0);
```

Comment:
You're potentially changing the passed in block state, and therefor changing vanilla mechanics. I get that the block state is already retrieved, but should comment that you're making that change since it's an overwrite.

Reply:
> You're potentially changing the passed in block state, and therefor changing vanilla mechanics.

I hope not, because `iblockstate` and `block` (`Block block = iblockstate.getBlock();`) are used here https://github.com/SpongePowered/SpongeCommon/blob/d1b8572f639d71267955becc8e2e263fa4f34770/src/main/java/org/spongepowered/common/mixin/core/world/MixinExplosion.java#L377-L378

to check if the block can be dropped. Either we can reuse iblockstate or we need to update iblockstate before L377 because that would be a bug in vanilla mc.

> but should comment that you're making that change since it's an overwrite.

Whoops, I missed that